### PR TITLE
WIP: Minimal cmake build script

### LINKED ISSUE #23: cmake script
It would be nice to include a `cmake` script (a `CMakeLists.txt` file) to make life easier for Windows users.   Since utf8proc is so simple, we can easily maintain both this and a `Makefile`.

### PR DESCRIPTION
**Pull request by tkelman**
The `disallow_intree_builds()` snippet is thanks to Tracy Wadleigh. Tested on Linux, will check with MSVC/MinGW next.

will close #23

### CODE REVIEW

**Comment by tkelman** [comment]
Tested on MSVC and MinGW (`-G"MSYS Makefiles"`) too. Builds either static or shared library, defaults to static unless you ask for `-DBUILD_SHARED_LIBS=ON`. Can still be spruced up for building the tests, adding an install target, etc.

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,6 @@
 *.dll
 *.dylib
 *.dSYM
-*.txt
```
Comment (stevengj) [inline .gitignore]: Add `data/*.txt` (or similar) back in, to exclude the Unicode data files that are downloaded there.

**Comment by stevengj** [comment]
Don't move `lump.txt` into `data/`, since it is documentation and not data.  Maybe rename it to `lump.md` and make a Markdown-formatted version.  (Also need to change mentions of `lump.txt` in the README and `utf8proc.[ch]` accordingly.)

**Comment by stevengj** [comment]
Also, can you add a test to the `.travis.yml` file?  (And we should test the Windows version with AppVeyor.)

**Comment by stevengj** [comment]
Also should have a `VERSION` and `SOVERSION`, right?

**Comment by tkelman** [comment]
> Also should have a `VERSION` and `SOVERSION`, right?

Yeah. Will be a little annoying to have to remember to keep the Makefile and CmakeLists synchronized here, won't it?

**Comment by kmsquire** [comment]
Why not just have CMakeLists.txt?

**Comment by tkelman** [comment]
If we make CMakeLists.txt do everything that the current Makefile does, sure. Should we just duplicate the info until that point?

**Comment by kmsquire** [comment]
I would think so, assuming @stevengj is okay with that plan.

**Comment by stevengj** [comment]
In the long term, we can think about moving over to `cmake` wholesale.  In the short term (utf8proc 1.2), I'd prefer to keep both and just duplicate the info.

**Comment by tkelman** [comment]
Will https://github.com/tkelman/libutf8proc/commit/ce053beeb63a805fb22e4b825259a9acc7de00f4 suffice for now? See the travis/appveyor logs for what that does to the output file names.

**Comment by stevengj** [comment]
@tkelman, it looks like this is calling `-Wl,-soname,libutf8proc.so.1.2.0`, which is wrong; the soname should be `-Wl,-soname,libutf8proc.so.1`.   `libutf8proc.so.1.2.0` is the library's "real name" — see [this HOWTO](http://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html).

**Comment by tkelman** [comment]
Only using `${SO_MAJOR}` seems to work. The "real name" is getting set to `libutf8proc.so.1.2-dev` but I guess that's alright for now?

**Comment by nalimilan** [comment]
The `-dev` suffix looks weird for a .so file name. Can't you get rid of it?

**Comment by stevengj** [comment]
@tkelman, it sounds like `cmake` wants both `SOVERSION` and `VERSION` to refer to the ABI version, not the API version.  hence `SOVERSION` should be `1` and `VERSION` should be `1.2`.

**Comment by tkelman** [comment]
Okay, I did that. Now if only Travis would run.

**Pull request merged**